packages/frontapp/src/app/components/BookInfo/index.tsx

```diff
@@ -472,6 +465,81 @@ const BookInfo: FC<IBookInfoProps> = ({
     setIsMaterialTakenByCurrentUser(false);
   };
 
+  const buttons = () => {
+    return userRole === RolesTypes.READER ? (
+      <>
+        {statusInfo?.person_id === id ? (
+          <WrapperButtons>
+            {!statusInfo ? (
+              <></>
+            ) : (
+              statusInfo?.status !== 'Free' && (
+                <>
+                  <StyledButton value="Return a book" onClick={retrieveBook} />
+                  <StyledButton
+                    value="Extend claim period"
+                    transparent
+                    onClick={prolongPeriod}
+                  />
+                </>
+              )
+            )}
+          </WrapperButtons>
+        ) : (
+          <WrapperButtons>
+            {statusInfo?.status !== 'Free' &&
+              (!isMaterialTakenByCurrentUser ? (
+                <StyledButton
+                  value="Notify when available"
+                  svgComponent={<EnableNotifySvg />}
+                  onClick={handleEnableNotifyButton}
+                />
+              ) : (
+                <StyledButton
+                  value="Cancel"
+                  svgComponent={<CancelNotifySvg />}
+                  transparent
+                  onClick={handleCancelNotifyButton}
+                />
+              ))}
+          </WrapperButtons>
+        )}
+        {statusInfo?.status === 'Free' ? (
+          <StyledButton
+            value="Claim a book"
+            svgComponent={<Claim />}
+            onClick={showClaimModal}
+          />
+        ) : null}
+      </>
+    ) : editing ? (
+      <WrapperButtons>
+        <StyledButton value="Save changes" onClick={editInformation} />
+        <StyledButton
+          value="Cancel changes"
+          transparent
+          onClick={discardChanges}
+        />
+      </WrapperButtons>
+    ) : (
+      <WrapperButtons>
+        <StyledButton
+          value="Edit information"
+          transparent
+          svgComponent={<Edit />}
+          onClick={handleEditBtn}
+        />
+        <StyledButton
+          value="Delete item"
+          transparent
+          secondary
+          svgComponent={<Remove />}
+          onClick={handleDeleteBtn}
+        />
+      </WrapperButtons>
+    );
+  };
+
```

Comment:
It's a bad practice to define component inside another component, react will re-create the component each time when `BookInfo` was changed, so the `buttons` component will be unmounted, removed from the DOM and then mounted again. Better to move it out either to another function inside `index.tsx` or to a different file/module

Reply:
We use these buttons only in this component, but we have to move them from top to bottom in laptop mode.
And i don't use this function as a component, but like this `{isLaptop ? buttons() : null}`.
Which is the best way to refactor it?

Reply:
> It's a bad practice to define component inside another component, react will re-create the component each time when `BookInfo` was changed, so the `buttons` component will be unmounted, removed from the DOM and then mounted again. Better to move it out either to another function inside `index.tsx` or to a different file/module

We use these buttons only in this component, but we have to move them from top to bottom in laptop mode.
And i don't use this function as a component, but like this `{isLaptop ? buttons() : null}`.
Which is the best way to refactor it?

Reply:
You can move it to a separate function within the file if it is the only place where you use it. Also, use functional component convention naming (`Buttons`)